vm/src/obj/objstr.rs

```diff
@@ -788,16 +791,21 @@ fn perform_format(
 // TODO: should with following format
 // class str(object='')
 // class str(object=b'', encoding='utf-8', errors='strict')
-fn str_new(vm: &mut VirtualMachine, args: PyFuncArgs) -> PyResult {
-    if args.args.len() == 1 {
-        return Ok(vm.new_str("".to_string()));
-    }
-
-    if args.args.len() > 2 {
-        panic!("str expects exactly one parameter");
+fn str_new(
```

Comment:
Now that methods exactly match their python names, should we do the same for free-standing functions? i.e. should this just be `str`?

Reply:
Maybe. I quite like the convention, it makes it a bit easier to know what's what. I think I'd call that ```new``` rather than ```str``` though.

Of course, that function could be a method.

Reply:
But it can't be a method called ```new``` because that clashes with the implementation of new in PyRef.

Reply:
I really like the idea of it being a `new` method though (feels more idiomatic). We can probably get rid of `PyRef::new` and have a generic `into_ref` on `PyValue`s.